### PR TITLE
Fix #2849 - Initial work of celery 5.0.0 alpha1 series by dropping python below 3.6 from marix & remove import from __future__

### LINKED ISSUE #2849: Remove Legacy Python (python2) support
Note that this will also remove `Jython` and `Pypy` support until they do not support python3.3+

### PR DESCRIPTION
**Pull request by auvipy**
work in progress for alpha one of celery 5.0.0.

this PR will drop python2 compat codes against v5-dev branch. All celery v5 work will be pushed against this branch and will be regularly synced with the master.

https://github.com/celery/celery/issues/2849

### CODE REVIEW

**Comment by auvipy** [comment]
As we discussed few months back @thedrow :D

File: README.rst [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 |build-status| |coverage| |license| |wheel| |pyversion| |pyimp| |ocbackerbadge| |ocsponsorbadge|
 
-:Version: 4.4.0rc3 (cliffs)
+:Version: 5.0.0a1 (cliffs)
```
Comment (thedrow) [inline README.rst]: Don't touch this.
Bumpversion is responsible for bumping the version.

Comment (auvipy) [reply README.rst]: Oh OK. I never used that tool so need your help on that :D also for the next releases willing to use something like bumpversion

**Comment by georgepsarakis** [comment]
Nice work @auvipy ! Will you also be addressing deprecations here, such as #5456 ?

**Comment by auvipy** [comment]
I will remove all the deprecation in another PR

**Comment by thedrow** [comment]
I agree.
However I really liked how bootstrap handled visibility major versions.
There's a long standing PR with all the changes but we never merge it until just before the release.
In fact, it should be a draft.

I'll create one.

**Comment by thedrow** [comment]
Oh @auvipy already created #5686.

But it's not a [draft PR](https://help.github.com/en/articles/changing-the-stage-of-a-pull-request).
Should we close it and open a new one.
Drafts are useful because it allows us to block merging until the release manager (me I guess? We should probably discuss that too) approves it is ready for final review and merge.

**Comment by auvipy** [comment]
dont worry that won't be merged in master, any way to make it as a draft?

**Comment by thedrow** [comment]
Unfortunately, no. You can only make a draft PR into a regular one for now.

**Pull request merged**